packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -279,6 +299,27 @@ export default util.createRule<Options, MessageIds>({
           return;
         }
 
+        if (isUnionOrIntersectionType(type)) {
```

Comment:
[Request] Tests still pass without this check. Is there any reason to include it? At this point I'd expect the type to be a union or intersection type, as `util.isNullableType(type, { allowUndefined: true })` was truthy.

```suggestion
```

[Tip] The type system may be upset after this about the type not being a `ts.UnionOrIntersectionType`. That's ok. I think an `as` assertion would be fine, or a type predicate just for this file.

Reply:
It's there so typescript will know it's an intersection (`type is ts.UnionOrIntersectionType` type guard)

Reply:
Yes, it tells TypeScript that it's an intersection - but _we_ know it's always an intersection. _We_ know the type is a union or intersection type because `util.isNullableType` returned true a few lines earlier. This _runtime_ code is only providing utility for the _type system_. That's a form of dead/unnecessary code.

I'm suggesting writing something like `(type as ts.UnionOrIntersectionType).types`.